### PR TITLE
Fixes designated initializer compiler warnings for iOS9

### PR DESCRIPTION
**Pull request by cryptojuice**
Overrides superclass designated initializer and throw's an NSException if the designated initializer should be used instead.

### CODE REVIEW

**Comment by cryptojuice** [comment]
This should fix #2301

**Comment by segiddins** [comment]
Awesome, thanks so much @cryptojuice! I'll merge once CI passes

File: Code/Search/RKSearchPredicate.m [hunk 1]
```diff
@@ -16,6 +16,22 @@ + (NSPredicate *)searchPredicateWithText:(NSString *)searchText type:(NSCompound
...
 
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+    @throw [NSException exceptionWithName:NSInternalInconsistencyException
+                                   reason:[NSString stringWithFormat:@"-initWithcoder: is not a valid initializer for the class %@, use designated initilizer -initWithSearchText:type:", NSStringFromClass([self class])]
+                                 userInfo:nil];
+    return [self init];
+}
+
+- (instancetype)initWithType:(NSCompoundPredicateType)type subpredicates:(NSArray<NSPredicate *> *)subpredicates
```
Comment (segiddins) [inline Code/Search/RKSearchPredicate.m]: please remove the generics syntax

**Comment by segiddins** [comment]
This is also causing a bunch of failures because of `RKMapperOperation`

**Comment by cryptojuice** [comment]
@segiddins looks like a bunch of test are calling init directly too.

**Comment by cryptojuice** [comment]
Hey @segiddins, what do you think about these changes? This branch now passes CI.

File: Code/CoreData/RKManagedObjectStore.m [hunk 2]
```diff
@@ -191,12 +207,6 @@ - (instancetype)initWithPersistentStoreCoordinator:(NSPersistentStoreCoordinator
     return self;
 }
 
-- (instancetype)init
```
Comment (segiddins) [inline Code/CoreData/RKManagedObjectStore.m]: why is this one removed?

**Comment by segiddins** [comment]
:+1: ace

File: Code/CoreData/RKManagedObjectStore.m (hunk 2 above)
Comment (cryptojuice) [inline Code/CoreData/RKManagedObjectStore.m]: We should be ok overriding init and calling initWithManagedObjectModel here. Reverting this change.

Comment (cryptojuice) [inline Code/CoreData/RKManagedObjectStore.m]: @segiddins I added the -init back, this should be ready for merge.

**Pull request merged**